ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -206,7 +206,7 @@ public BlobStore(ReplicaId replicaId, StoreConfig config, ScheduledExecutorServi
     ttlUpdateBufferTimeMs = TimeUnit.SECONDS.toMillis(config.storeTtlUpdateBufferTimeSeconds);
     errorCount = new AtomicInteger(0);
     currentState = ReplicaState.OFFLINE;
-    remoteTokenTracker = replicaId == null ? null : new RemoteTokenTracker(replicaId);
+    remoteTokenTracker = replicaId == null ? null : new RemoteTokenTracker(replicaId, taskScheduler, factory);
```

Comment:
just curious, this remote token tracker is in the memory of the sender - the server that sends blobs right ?
and with this change you are persisting that token. So if B pulls from A, then A persists its remoteToken corresponding to B?

Reply:
Correct.